File: 1528-shuffle-string/1528-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        StringBuilder res = new StringBuilder(s);
        for (int i = 0; i < indices.length; i++) {
            res.setCharAt(indices[i], s.charAt(i));
        }
        return res.toString();
    }
}

File: 1528-shuffle-string/1528-shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    let str = [];
    for(let i=0;i<indices.length;i++){
        str[indices[i]]=s[i];
    }
    return str.join("");
};